### PR TITLE
feat(lang/c-cpp): add some differences

### PR DESCRIPTION
**Pull request by CoelacanthusHex**
https://github.com/OI-wiki/gitment/issues/398#issuecomment-767988680

### CODE REVIEW

**Review by Enter-tainer — DISMISSED** [review]

**Review by Xeonacid — COMMENTED** [review]
行末句号（

File: docs/lang/c-cpp.md [hunk 1]
```diff
@@ -39,3 +39,19 @@ const 在 C 中只有限定变量不能修改的功能，而在 C++ 中，由于
...
+
+C++ 风格单行注释 `//`，C 于 C99 前不支持
+
+## 变量声明
+
+C99 前，C 的变量声明必须位于语句块开头，C++ 和 C99 后无此限制
+
+## 可变长数组
+
+C99 后 C 语言支持 VLA （可变长数组），C++ 始终不支持
```
Comment (ksyx) [inline docs/lang/c-cpp.md:53]: 非标准

File: docs/lang/c-cpp.md [hunk 2]
```diff
@@ -39,3 +39,19 @@ const 在 C 中只有限定变量不能修改的功能，而在 C++ 中，由于
 C++ 中新增了 `new` 和 `delete` 关键字用来在“自由存储区”上分配空间，这个自由存储区可以是堆也可以是静态存储区，他们是为了配合“类”而出现的。其中 `delete[]` 还能够直接释放动态数组的内存，非常方便。`new` 和 `delete` 关键字会调用类型的构造函数和析构函数，相比 C 中的 `malloc()`、`realloc()`、`free()` 函数，他们对类型有更完善的支持，但是效率不如 C 中的这些函数。
 
 简而言之，如果你需要动态分配内存的对象是基础类型或他们的数组，那么你可以使用 `malloc()` 进行更高效的内存分配；但如果你新建的对象是非基础的类型，那么建议使用 `new` 以获得安全性检查。值得注意的是尽管 `new` 和 `malloc()` 都是返回指针，但是 `new` 出来的指针 **只能** 用 `delete` 回收，而 `malloc()` 出来的指针也只能用 `free()` 回收，否则会有内存泄漏的风险。
+
+## 注释
+
+C++ 风格单行注释 `//`，C 于 C99 前不支持
+
+## 变量声明
```
Comment (ksyx) [inline docs/lang/c-cpp.md:47]: 感觉下面这几个是三级标题

File: docs/lang/c-cpp.md [hunk 3]
```diff
@@ -39,3 +39,19 @@ const 在 C 中只有限定变量不能修改的功能，而在 C++ 中，由于
...
+
+C99 前，C 的变量声明必须位于语句块开头，C++ 和 C99 后无此限制
+
+## 可变长数组
+
+C99 后 C 语言支持 VLA （可变长数组），C++ 始终不支持
+
+## 结构体初始化
+
+C99 后 C 语言支持结构体的[指派符初始化](https://en.cppreference.com/w/c/language/struct_initialization)，C++ 直到 C++20 才支持有顺序要求的指派符初始化（C 的沒有顺序要求也不要求必须都初始化）
```
Comment (ksyx) [inline docs/lang/c-cpp.md:57]:
```suggestion
C99 后 C 语言支持结构体的 [指派符初始化](https://en.cppreference.com/w/c/language/struct_initialization)，C++ 直到 C++20 才支持有顺序要求的指派符初始化（C 语言没有顺序要求也不要求必须都初始化）
```

File: docs/lang/c-cpp.md (hunk 1 above)
Comment (CoelacanthusHex) [reply docs/lang/c-cpp.md:53]: ？

**Comment by CoelacanthusHex** [comment]
@Xeonacid fixed

**Review by Xeonacid — APPROVED** [review]

File: docs/lang/c-cpp.md [hunk 4]
```diff
@@ -39,3 +39,19 @@ const 在 C 中只有限定变量不能修改的功能，而在 C++ 中，由于
...
+
+## 可变长数组
+
+C99 后 C 语言支持 VLA （可变长数组），C++ 始终不支持。
+
+## 结构体初始化
+
+C99 后 C 语言支持结构体的 [指派符初始化](https://en.cppreference.com/w/c/language/struct_initialization)，C++ 直到 C++20 才支持有顺序要求的指派符初始化（C 语言没有顺序要求也不要求必须都初始化）。
+
+## 注释
```
Comment (ksyx) [inline docs/lang/c-cpp.md:55]:
```suggestion
## 注释语法
```
not *footnote*

**Review by ksyx — APPROVED** [review]
thx

**Review by Xeonacid — APPROVED** [review]

**Pull request merged**